### ISSUE TITLE
AWS Cloud Map resource tagging

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#13696]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Cloud Map (AKA ServiceDiscovery) namespaces and services now support resource tags.

### New or Affected Resource(s)

* [`aws_service_discovery_http_namespace`](https://www.terraform.io/docs/providers/aws/r/service_discovery_http_namespace.html)
* [`aws_service_discovery_private_dns_namespace`](https://www.terraform.io/docs/providers/aws/r/service_discovery_private_dns_namespace.html)
* [`aws_service_discovery_public_dns_namespace`](https://www.terraform.io/docs/providers/aws/r/service_discovery_public_dns_namespace.html)
* [`aws_service_discovery_service`](https://www.terraform.io/docs/providers/aws/r/service_discovery_service.html)

### Potential Terraform Configuration

```hcl
resource "aws_service_discovery_private_dns_namespace" "example" {
  tags = {
    Environment = "test"
  }
}
```

### References

[Developer Guide](https://docs.aws.amazon.com/cloud-map/latest/dg/using-tags.html).

Requires [AWS SDK v1.31.13](https://github.com/aws/aws-sdk-go/releases/tag/v1.31.13):
* https://github.com/terraform-providers/terraform-provider-aws/pull/13459

**Comment by bflad** [comment]
keyvaluetags portion submitted here: https://github.com/terraform-providers/terraform-provider-aws/pull/13732

**Comment by bflad** [comment]
Support for tagging functionality with the Cloud Map resources has been merged and will release with version 2.67.0 of the Terraform AWS Provider, later this week. Thanks to @ewbankkit for the implementation. 👍
